apollo-router/src/axum_factory/axum_http_server_factory.rs

```diff
@@ -329,7 +329,16 @@ async fn handle_graphql(
     service: router::BoxService,
     http_request: Request<Body>,
 ) -> impl IntoResponse {
-    match service.oneshot(http_request.into()).await {
+    let request: router::Request = http_request.into();
+    let context = request.context.clone();
+
+    let res = service.oneshot(request).await;
+    let dur = context.busy_time_ns().await;
+    let processing_seconds = dur as f64 / 1_000_000_000f64;
```

Comment:
NIT Presumably not returning a duration because it would be higher overhead... If not using duration we could have inline function on BusyTime that does the calculation.

As noted elsewhere, I wonder if "seconds" is the correct unit here.

Reply:
good point, `Duration` has the[as_secs_f64](https://doc.rust-lang.org/stable/std/time/struct.Duration.html#method.as_secs_f64) method

Reply:
fixed in 9804f0db